### PR TITLE
Bl8517 defect alert screen

### PR DESCRIPTION
**Pull request by karlbeckettbjss**

### CODE REVIEW

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java [hunk 1]
```diff
@@ -1635,18 +1640,20 @@ private void markAsRepaired(String defect) {
...
      * cucumber steps, the original steps are detailed below.
-     * @param defectType        The defect type, must be "Advisory", "PRS" or "Failure"
-     * @param defect            The defect
-     * @param updatedComment    The updated comment
-     * @param isDangerous       Whether this defect should be marked as dangerous
+     * @param defectType            The defect type, must be "Advisory", "PRS" or "Failure"
+     * @param defect                The defect
+     * @param updatedComment        The updated comment
+     * @param isDangerous           Whether this defect should be marked as dangerous
+     * @param isFirstUseDateWarning Whether this defect should be marked as dangerous
```
Comment (ibrahimjassat) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: Does not seem right

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java [hunk 2]
```diff
@@ -1689,7 +1696,26 @@ private void addDefectFromBrowse(String defectType, String category, Optional<St
 
         // Add the defect
         addDefectAndReturnToResults(DefectJourney.AddFromBrowse, defect, DefectType.fromString(defectType),
-                comment, false, "Defects");
+                comment, false, false,"Defects");
```
Comment (ibrahimjassat) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: missing a space

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java [hunk 3]
```diff
@@ -1689,7 +1696,26 @@ private void addDefectFromBrowse(String defectType, String category, Optional<St
...
+     * @param comment           The comment to use
+     */
+    private void addDefectFromBrowseAlert(String defectType, String category, Optional<String> subcategory,
+                                     Optional<String> subSubCategory, String defect, String comment) {
+        // Browse to the desired defect
+        browseForDefect(category, subcategory, subSubCategory);
+
+        // Add the defect
+        addDefectAndReturnToResults(DefectJourney.AddFromBrowse, defect, DefectType.fromString(defectType),
+                comment, false, true,"Defects");
```
Comment (ibrahimjassat) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: same again here

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java [hunk 4]
```diff
@@ -1804,7 +1831,7 @@ private void addDefectFromSearch(String defectType, String defect, String commen
 
         //Add the defect and return to results
         addDefectAndReturnToResults(DefectJourney.AddFromSearch, defect, DefectType.fromString(defectType),
-                comment, false, "Search for a defect");
+                comment, false, false,"Search for a defect");
```
Comment (ibrahimjassat) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: same again here

**Review by ibrahimjassat — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java (hunk 1 above)
Comment (karlbeckettbjss) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: Fixed

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java (hunk 2 above)
Comment (karlbeckettbjss) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: Fixed

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java (hunk 3 above)
Comment (karlbeckettbjss) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: Fixed

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java (hunk 4 above)
Comment (karlbeckettbjss) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/TesterDoesStepDefinitions.java]: Fixed

**Review by ibrahimjassat — APPROVED** [review]

**Comment by karlbeckettbjss** [comment]
This will be merged once the feature has been delivered

**Pull request merged**